library/src/interactor_impl.cxx

```diff
@@ -193,7 +198,16 @@ class interactor_impl::internals
         self->Window.PrintSceneDescription(log::VerboseLevel::INFO);
         break;
       default:
-        if (keySym == "Left")
+
+        if (keyCode == '2' || keySym == "KP_Down")
+        {
+          self->VTKInteractor->SetInteractorStyle(self->Style2D);
+        }
+        else if (keyCode == '3' || keySym == "KP_Next")
+        {
+          self->VTKInteractor->SetInteractorStyle(self->Style3D);
```

Comment:
this should be controlled by an option

Reply:
Yeah, I will do that but there is a misconception somewhere IMHO because:
 - InteractorStyles are own by the interactor:
 - Options are managed by the window. But window doesn't have access to either interactor or engine (which owns both window and interactor).
Am I missing a link?

Not directly related:
  - options class should be free of "f3d" internals. We should have a derived f3doptions class (or so) that performs the declaration/initialization of the f3d specific options.

Reply:
> Options are managed by the window. But window doesn't have access to either interactor or engine (which owns both window and interactor). Am I missing a link?

See how it is done for Trackball. That being said, adding a logic for interactor specific options could make sense in the long run. Right now there are not a lot of such options, and an interactor may not always be available, so it is a bit easier this way.


> options class should be free of "f3d" internals. We should have a derived f3doptions class (or so) that performs the declaration/initialization of the f3d specific options.

You mean options.h ? There is only a pointer to an internal class, but it could be done indeed. not sure this is worth it. Please open a dedicate issue to discuss this.

>    to be more aligned with the lib approach, the mapping between key code/sym and options should be performed at a higher level, like in the f3dstart class.

Definitely, this is something which was planned but I did not get around to do it. It would be nice to track it in a dedicated issue.

Thanks for your suggestions !

Reply:
> See how it is done for Trackball.
I saw that but it is different. For trackball, option is set to to the vtkF3DRenderer, and then the InteractorStyle3D check its value because it has access to the renderer.
Here, the interactor styles are managed by the interactor_impl internal class. 
I think window_impl::UpdateDynamicOptions should be able to access to the interactor(_impl) instance own by the engine (which also is not visible from window_impl). If you agree with this statement, I can make changes to allow this access.

Reply:
Yes, that may be needed.